### ISSUE TITLE
Rofi window mode shows wrong desktop names in i3 (4.16)

### ISSUE THREAD

**Issue by b1scu1t** [DaveDavenport/rofi#901]
:exclamation:
First read the [guidelines](https://github.com/DaveDavenport/rofi/blob/next/.github/CONTRIBUTING.md)!
:exclamation:

## Version

Output of `rofi -v`

Version: 1.5.2

## Configuration

Output of `rofi -help` (in a [gist](https://gist.github.com/))

[Rofi configuration](https://gist.github.com/b1scu1t/892fc69febc0f650133815a1eabc607b/bcd40e622a511996d093166de83bd7296c351135)
## Launch Command

The commandline used to launch **rofi**

`rofi -show window -show-icons`

## Steps to reproduce

* Step 1
Launch i3-gaps or vanilla i3 (ver. 4.16) and have 4 workspaces(ws); one has Firefox (ws2) and the other 3 (ws1,ws3,ws4) have Xfce4-Terminal open.

* Step 2
Run `rofi -show window -show-icons`

## What behavior you see

* - Rofi doesn't show the correct desktop names. It displays a mixture of 'desktop' and 'workspace' names in the list, with the wrong workspace numbers tied to each window.

![image](https://user-images.githubusercontent.com/41207956/51227788-3fbb3680-192c-11e9-9001-b689581539f5.png)

## What behaviour you expect to see

* - Rofi would normally display the workspace names specified in the i3-config file. 

The specified workspace names have glyphs provided by `Inconsolata Nerd Font Mono`.
The i3-bar and the i3-workspaces-plugin (on xfce4-panel) display the names correctly.

![image](https://user-images.githubusercontent.com/41207956/51229215-f1a93180-1931-11e9-9680-c26a6fa3b8c7.png)

Output of `i3-msg -t get_workspaces`

`[{"num":1,"name":"1 ","visible":false,"focused":false,"rect":{"x":0,"y":0,"width":1366,"height":743},"output":"LVDS1","urgent":false},{"num":2,"name":"2 ","visible":true,"focused":true,"rect":{"x":0,"y":0,"width":1366,"height":743},"output":"LVDS1","urgent":false},{"num":3,"name":"3 ","visible":false,"focused":false,"rect":{"x":0,"y":0,"width":1366,"height":743},"output":"LVDS1","urgent":false},{"num":4,"name":"4 ","visible":false,"focused":false,"rect":{"x":0,"y":0,"width":1366,"height":743},"output":"LVDS1","urgent":false}]`

Screenshot of my i3 config
![screenshot_2019-01-16_01-42-18](https://user-images.githubusercontent.com/41207956/51228770-2b793880-1930-11e9-8a84-a636513b06f7.png)

My i3 config
[config.txt](https://github.com/DaveDavenport/rofi/files/2762846/config.txt)

Thank you for your time.

**Comment by b1scu1t** [comment]
Output of `xprop -root`
```
_NET_ACTIVE_WINDOW(WINDOW): window id # 0x1a00067
AT_SPI_BUS(STRING) = "unix:abstract=/tmp/dbus-BVTz5nu8Tv,guid=fcdc3a2a0533204bd1b7b7005c3f113f"
XKLAVIER_ALLOW_SECONDARY(INTEGER) = 0
GDK_VISUALS(INTEGER) = 212, 267
_MIT_SCREEN_SAVER_ID(PIXMAP): pixmap id # 0x600000
_NET_DESKTOP_VIEWPORT(CARDINAL) = 0, 0, 0, 0, 0, 0, 0, 0
_NET_DESKTOP_NAMES(UTF8_STRING) = "Workspace 1", "2 ", "10 﫸", "10 﫸", ""
_NET_NUMBER_OF_DESKTOPS(CARDINAL) = 4
_NET_CURRENT_DESKTOP(CARDINAL) = 0
I3_CONFIG_PATH(UTF8_STRING) = "/home/biscuith/.config/i3/config"
I3_PID(CARDINAL) = 1043
I3_SOCKET_PATH(UTF8_STRING) = "/run/user/1000/i3/ipc-socket.1043"
_NET_CLIENT_LIST(WINDOW): window id # 0x1a00067, 0x1e00010, 0x1a000fc, 0x1a0013d, 0x1e0007e
_NET_CLIENT_LIST_STACKING(WINDOW): window id # 0x1a000fc, 0x1a0013d, 0x1e00010, 0x1e0007e, 0x1a00067
_NET_SUPPORTED(ATOM) = _NET_SUPPORTED, _NET_SUPPORTING_WM_CHECK, _NET_WM_NAME, _NET_WM_VISIBLE_NAME, _NET_WM_MOVERESIZE, _NET_WM_STATE_STICKY, _NET_WM_STATE_FULLSCREEN, _NET_WM_STATE_DEMANDS_ATTENTION, _NET_WM_STATE_MODAL, _NET_WM_STATE_HIDDEN, _NET_WM_STATE_FOCUSED, _NET_WM_STATE, _NET_WM_WINDOW_TYPE, _NET_WM_WINDOW_TYPE_NORMAL, _NET_WM_WINDOW_TYPE_DOCK, _NET_WM_WINDOW_TYPE_DIALOG, _NET_WM_WINDOW_TYPE_UTILITY, _NET_WM_WINDOW_TYPE_TOOLBAR, _NET_WM_WINDOW_TYPE_SPLASH, _NET_WM_WINDOW_TYPE_MENU, _NET_WM_WINDOW_TYPE_DROPDOWN_MENU, _NET_WM_WINDOW_TYPE_POPUP_MENU, _NET_WM_WINDOW_TYPE_TOOLTIP, _NET_WM_WINDOW_TYPE_NOTIFICATION, _NET_WM_DESKTOP, _NET_WM_STRUT_PARTIAL, _NET_CLIENT_LIST, _NET_CLIENT_LIST_STACKING, _NET_CURRENT_DESKTOP, _NET_NUMBER_OF_DESKTOPS, _NET_DESKTOP_NAMES, _NET_DESKTOP_VIEWPORT, _NET_ACTIVE_WINDOW, _NET_CLOSE_WINDOW, _NET_MOVERESIZE_WINDOW
_NET_WM_NAME(UTF8_STRING) = "i3"
_NET_SUPPORTING_WM_CHECK(WINDOW): window id # 0x200063
RESOURCE_MANAGER(STRING) = "*.color0:\t#292929\n*.color1:\t#c74a4b\n*.color10:\t#ddff6e\n*.color11:\t#ffff8d\n*.color12:\t#93ffff\n*.color13:\t#ff96f7\n*.color14:\t#85f3ff\n*.color15:\t#ffffff\n*.color2:\t#a5c264\n*.color3:\t#ffd280\n*.color4:\t#82ccbf\n*.color5:\t#bf84b4\n*.color6:\t#77b4cc\n*.color7:\t#e3e3e3\n*.color8:\t#3d3d3d\n*.color9:\t#ff4e50\nURxvt.background:\t[85]#292929\nURxvt.cursorColor:\t#e3e3e3\nURxvt.depth:\t32\nURxvt.font:\txft:Inconsolata Nerd Font Mono:pixelsize=12:antialias=false, xft:xos4 Terminus:pixelsize=12:antialias=false\nURxvt.foreground:\t#e3e3e3\nURxvt.letterSpace:\t0\nURxvt.matcher.button:\t1\nURxvt.perl-ext-common:\tdefault,matcher,resize-font\nURxvt.scrollBar:\tfalse\nURxvt.url-launcher:\t/usr/bin/xdg-open\nXft.antialias:\t1\nXft.hinting:\t0\nXft.hintstyle:\thintnone\nXft.rgba:\tnone\nXft.dpi:\t96\nXcursor.theme:\tAdwaita\nXcursor.size:\t16\nXcursor.theme_core:\t1\n"
XFree86_DDC_EDID1_RAWDATA(INTEGER) = 0, -1, -1, -1, -1, -1, -1, 0, 48, -28, 53, 3, 0, 0, 0, 0, 0, 21, 1, 3, -128, 31, 17, 120, -22, -47, 5, -107, 88, 87, -113, 41, 32, 80, 84, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 18, 27, 86, 104, 80, 0, 18, 48, 32, 32, 53, 0, 54, -82, 16, 0, 0, 25, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -2, 0, 76, 71, 32, 68, 105, 115, 112, 108, 97, 121, 10, 32, 32, 0, 0, 0, -2, 0, 76, 80, 49, 52, 48, 87, 72, 50, 45, 84, 76, 70, 49, 0, -43
_XKB_RULES_NAMES(STRING) = "evdev", "pc105", "us", "", ""
XFree86_has_VT(INTEGER) = 1
XFree86_VT(INTEGER) = 1
Xorg_Seat(STRING) = "seat0"

```
I've got the same workspaces (1-4); two Firefox windows on workspace 2 and Terminals on 1,3 and 4.

**Comment by DaveDavenport** [comment]
These names are used by rofi to determine the desktop name:
```

_NET_DESKTOP_NAMES(UTF8_STRING) = "Workspace 1", "2 ", "10 﫸", "10 﫸", ""
```

Are these correct to what you expect?

**Comment by b1scu1t** [comment]
No, I was expecting this:
`_NET_DESKTOP_NAMES(UTF8_STRING) = "1 ","2 ","3 ","4 ",""`

I'm not sure if the glyph icons are visible on your side. 

Here's a screenshot of my panel, showing the correct desktop names.
![image](https://user-images.githubusercontent.com/41207956/51265685-b0e00580-198f-11e9-98a8-1931a697bb20.png)

These are the assigned desktop names in i3.
![image](https://user-images.githubusercontent.com/41207956/51265972-6448fa00-1990-11e9-8b59-db804ddb6c54.png)

It seems like `xprop -root` is returning the wrong desktop names.
The `i3-msg -t get_workspaces` command get the correct ones.
![image](https://user-images.githubusercontent.com/41207956/51266376-7ecfa300-1991-11e9-9a88-e60364182f77.png)

Rofi worked well in the past, maybe it's a regression on i3(4.16)? 
I haven't seen any complaints elsewhere, so probably there may be something influencing `xprop -root.`

**Comment by DaveDavenport** [comment]
That looks like a regression in i3, because that is the one setting the _NET_DESKTOP_NAMES properties.

@Airblader Any ideas on this?

**Comment by b1scu1t** [comment]
@Airblader  @DaveDavenport 
It looks like I found the culprit; its the `xfsettingsd` daemon. 
Rofi without the `xfsettingsd` daemon.
![rofi_works](https://user-images.githubusercontent.com/41207956/51276159-50aa8d00-19aa-11e9-8bf9-0c12d6a529aa.png)

I use that program to set or reload GTK themes on the fly with `xfce4-settings-manager` or `wpg`.
 I had it auto-start from the i3 config file in the i3-session, the xfce4-session always starts it automatically.

I checked `xfconf` and it looks like `xfsettingsd` sets the desktop names, to be used by `xfwm4` (which isn't even installed). The `xfwm4` entry would re-appear after deletion, with the same bad desktop names, as soon as I switch the workspace in i3.

Screenshot of the xfconf editor on the `xfwm4` entry. (Note the desktop names.)
![rofi_culprit](https://user-images.githubusercontent.com/41207956/51275456-68811180-19a8-11e9-990b-28554834996a.png)

Screenshot of Rofi with `xfsettingsd` on. (Note the desktop names.)
![image](https://user-images.githubusercontent.com/41207956/51276400-0675db80-19ab-11e9-8ca9-265fa8a3dc13.png)
